### PR TITLE
DNS-138 Define & Rollout the infrastructure to all stages

### PR DESCRIPTION
**Pull request by dandrejevic-ionos**
Pull request created in: https://jira.sso.ionos.cloud/browse/DNS-138

### CODE REVIEW

File: .github/workflows/deploy-manual.yaml [hunk 1]
```diff
@@ -2,12 +2,132 @@ name: Deploy (manual)
...
+
+concurrency:
+  group: paasdns-${{ github.event.inputs.environment }}
+  cancel-in-progress: false
+
+env:
+  ENVIRONMENT: ${{ github.event.inputs.environment }}
+  LOCATION: ${{ github.event.inputs.location }}
+  IONOS_TOKEN: ${{ secrets.IONOS_TOKEN }}
+  S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
```
Comment (mspoeri) [inline .github/workflows/deploy-manual.yaml:51]: Are there currently plans how we could separate the dev from the prod bucket?

**Review by mspoeri — COMMENTED** [review]
The rest looks good to me

File: .github/workflows/deploy-manual.yaml (hunk 1 above)
Comment (dandrejevic-ionos) [reply .github/workflows/deploy-manual.yaml:51]: That actually was my question as well.

Initially, I wanted to create bucket per environment... but we ended up with a single bucket (paasdns-bucket), but with separate environment directories (dev/stage/qa/prod) and separate location directories inside them (de-fra/de-txl/...).

So, currently, it looks like this for instance - `paasdns-bucket > dev > de-fra > terraform.tfstate`.

There was also a mention that we are going to use one bucket for non-prod environments (qa/dev/stage) and a separate one for production.

In any case, there is a way to specify bucket name in the [S3_BUCKET environment variable](https://github.com/ionos-cloud/paas-dns-infrastructure/blob/DNS-138-define-and-rollout-the-infrastructure-to-all-stages/main.go#L36), but we are relying on the default value for now.

Comment (mspoeri) [reply .github/workflows/deploy-manual.yaml:51]: Yes that was one topic we discussed that we want(or even must) to have an isolated contract for production and with this also a dedicated S3 within the production contract for the production cluster.

So I would recommend for now to exclude this from this task and ticket and pick this up later so we can now go on with the other tasks

**Pull request merged**